dotnet-desktop-guide/net/winforms/controls-design/usercontrol-guidelines.md

```diff
@@ -0,0 +1,51 @@
+---
+title: User control guidelines
+description: This article offers practical advice on how to design your user control for Windows Forms.
+ms.date: 08/08/2023
+ms.topic: overview
+dev_langs:
+  - "csharp"
+  - "vb"
+f1_keywords: 
+  - "UserControl"
+helpviewer_keywords: 
+  - "events [Windows Forms], defining within Windows Forms custom controls"
+  - "custom controls [Windows Forms], events using code"
+  - "property [Windows Forms], property changed events"
+  - "property [Windows Forms], designer"
+---
+
+# Guidelines to designing user controls (Windows Forms .NET)
+
+This article provides guidelines for creating user controls. We recommend that you follow these guidelines to make sure that you design a user control that's consistent with other Windows Forms controls.
+
+[!INCLUDE [desktop guide under construction](../../includes/desktop-guide-preview-note.md)]
+
+## Defining events
+
+Events commonly communicate state change and alert you to how the user is interacting with a Windows Forms control. For more information about events and delegates, see [Handle and raise events](/dotnet/standard/events/index).
+
+When defining your own events, follow these suggestions:
+
+- Use the <xref:System.EventHandler> event delegate when you define an event that doesn't have any associated data. Use the <xref:System.EventHandler%601> event delegate when you do have associated data.
+- Pass the control instance as the `sender` parameter.
+- Derive from <xref:System.EventArgs> and extend it with your data, when you raise an event with custom data.
+- Create a method named `On{EventName}` that raises the event, which is marked as `protected` and `virtual` (in C#) or `Protected` and `Overridable` (in Visual Basic).
+
+:::code language="csharp" source="./snippets/usercontrol-guidelines/csharp/UserControl1.cs" id="property_changed":::
+:::code language="vb" source="./snippets/usercontrol-guidelines/vb/UserControl1.vb" id="property_changed":::
+
+## Property changed events
+
+If you want your control to send notifications when a property changes, define an event named `{PropertyName}Changed`. This is the naming convention used in Windows Forms. The associated event delegate type for property-changed events is <xref:System.EventHandler>, and the event data type is <xref:System.EventArgs>. The base class <xref:System.Windows.Forms.Control> defines many property-changed events, such as <xref:System.Windows.Forms.Control.BackColorChanged>, <xref:System.Windows.Forms.Control.BackgroundImageChanged>, <xref:System.Windows.Forms.Control.FontChanged>, <xref:System.Windows.Forms.Control.LocationChanged>.  
```

Comment:
We should add a comment here, that this is important, if a property should be capable to bind bi-directional to a data source.